src/Nancy/HttpLink.cs

```diff
@@ -0,0 +1,262 @@
+        /// <param name="title">The optional title parameter is used to label the destination of a link such that it can be used as a human-readable identifier (e.g., a menu entry) in the language indicated by the HTTP Content-Language header (if present).</param>
+        /// <exception cref="System.ArgumentNullException"><paramref name="targetUri" /> or <paramref name="relation" /></exception>
+        public HttpLink(Uri targetUri, string relation, string type, string title)
+        {
+            if (targetUri == null)
+            {
+                throw new ArgumentNullException("targetUri");
+            }
+
+            if (relation == null)
+            {
+                throw new ArgumentNullException("relation");
+            }
+
+            this.Add("rel", HttpLinkRelation.Parse(relation));
+            this.targetUri = targetUri;
+
+            if (!string.IsNullOrWhiteSpace(type))
+            {
+                this.Add("type", new MediaRange(type));
+            }
+
+            if (!string.IsNullOrEmpty(title))
+            {
+                this.Add("title", title);
+            }
+        }
+
+        /// <summary>
+        /// The relation that identifies the semantics of the link.
+        /// </summary>
+        public HttpLinkRelation Relation
+        {
+            get { return this.GetParamValue<HttpLinkRelation>("rel"); }
+        }
+
+        /// <summary>
+        /// Gets the target URI.
+        /// </summary>
+        public Uri TargetUri
+        {
+            get { return this.targetUri; }
+        }
+
+        /// <summary>
+        /// The optional title parameter is used to label the destination of a link such that it can be used as a human-readable identifier (e.g., a menu entry) in the language indicated by the HTTP Content-Language header (if present).
+        /// </summary>
+        public string Title
+        {
+            get { return this.GetParamValue<string>("title"); }
+        }
+
+        /// <summary>
+        /// The optional type parameter is a hint indicating what the media type of the result of dereferencing the link should be. Note that this is only a hint; for example, it does not override the HTTP Content-Type header of a HTTP response obtained by actually following the link.
+        /// </summary>
+        public MediaRange Type
+        {
+            get { return this.GetParamValue<MediaRange>("type"); }
+        }
+
+        /// <summary>
+        /// Indicates whether the current object is equal to another object of the same type.
+        /// </summary>
+        /// <param name="other">An object to compare with this object.</param>
+        /// <returns>
+        /// true if the current object is equal to the <paramref name="other" /> parameter; otherwise, false.
+        /// </returns>
+        bool IEquatable<HttpLink>.Equals(HttpLink other)
```

Comment:
@khellang It's just a personal preference to not have several overloads publicly available that do the exact same thing.